compiler/test/stdlib/bytes.test.gr

```diff
@@ -0,0 +1,191 @@
+let bytes = Bytes.make(4)
+Bytes.setFloat32(0, 42.0f, bytes)
+assert Bytes.getFloat32(0, bytes) == 42.0f
+
+
+// Bytes.setInt64, Bytes.getInt64
+let bytes = Bytes.make(8)
+Bytes.setInt64(0, 0x7fffffffffffffffL, bytes)
+assert Bytes.getInt64(0, bytes) == 0x7fffffffffffffffL
+
+
+// Bytes.setFloat64, Bytes.getFloat64
+let bytes = Bytes.make(8)
+Bytes.setFloat64(0, 42.0d, bytes)
+assert Bytes.getFloat64(0, bytes) == 42.0d
+
+
+// Bytes.copy
+let bytes0 = Bytes.make(8)
+assert Bytes.length(bytes0) == 8
+for (let mut i = 0; i < 8; i+= 1) {
+  Bytes.setInt8(i, Int32.fromNumber(i), bytes0)
+}
+let bytes1 = Bytes.copy(bytes0);
+assert Bytes.length(bytes1) == Bytes.length(bytes0)
+for (let mut i = 0; i < 8; i += 1) {
+  assert Bytes.getInt8S(i, bytes1) == Bytes.getInt8S(i, bytes0)
+}
+
+
+// Bytes.sub
+let bytes2 = Bytes.sub(0, 8, bytes1)
+assert Bytes.length(bytes2) == Bytes.length(bytes1)
+for (let mut i = 0; i < 8; i += 1) {
+  assert Bytes.getInt8S(i, bytes2) == Bytes.getInt8S(i, bytes1)
+}
+let bytes3 = Bytes.sub(4, 2, bytes2)
+assert Bytes.length(bytes3) == 2
+for (let mut i = 0; i < Bytes.length(bytes3); i += 1) {
+  assert Bytes.getInt8S(i, bytes3) == Bytes.getInt8S(i + 4, bytes2)
+}
+let bytes4 = Bytes.sub(1, 6, bytes2)
+assert Bytes.length(bytes4) == 6
+for (let mut i = 0; i < Bytes.length(bytes4); i += 1) {
+  assert Bytes.getInt8S(i, bytes4) == Bytes.getInt8S(i + 1, bytes2)
+}
+
+
+// Bytes.extend
+let bytes5 = Bytes.extend(0, 0, bytes0)
+assert Bytes.length(bytes5) == Bytes.length(bytes0)
+for (let mut i = 0; i < Bytes.length(bytes5); i += 1) {
+  assert Bytes.getInt8S(i, bytes5) == Bytes.getInt8S(i, bytes0)
+}
+let bytes5 = Bytes.extend(1, 1, bytes0)
+assert Bytes.length(bytes5) == (Bytes.length(bytes0) + 2)
+for (let mut i = 0; i < Bytes.length(bytes0); i += 1) {
+  assert Bytes.getInt8S(i + 1, bytes5) == Bytes.getInt8S(i, bytes0)
+}
+let bytes5 = Bytes.extend(3, 0, bytes0)
+assert Bytes.length(bytes5) == 11
+for (let mut i = 0; i < Bytes.length(bytes0); i += 1) {
+  assert Bytes.getInt8S(i + 3, bytes5) == Bytes.getInt8S(i, bytes0)
+}
+let bytes5 = Bytes.extend(0, 3, bytes0)
+assert Bytes.length(bytes5) == 11
+for (let mut i = 0; i < Bytes.length(bytes0); i += 1) {
+  assert Bytes.getInt8S(i, bytes5) == Bytes.getInt8S(i, bytes0)
+}
+let bytes5 = Bytes.extend(-4, 0, bytes0)
+assert Bytes.length(bytes5) == 4
+for (let mut i = 0; i < Bytes.length(bytes5); i += 1) {
+  assert Bytes.getInt8S(i, bytes5) == Bytes.getInt8S(i + 4, bytes0)
+}
+let bytes5 = Bytes.extend(0, -4, bytes0)
+assert Bytes.length(bytes5) == 4
+for (let mut i = 0; i < Bytes.length(bytes5); i += 1) {
+  assert Bytes.getInt8S(i, bytes5) == Bytes.getInt8S(i, bytes0)
+}
+let bytes5 = Bytes.extend(-4, -4, bytes0)
+assert Bytes.length(bytes5) == 0
+
+
+// Bytes.move
+let srcBytes = Bytes.copy(bytes0)
+let dstBytes = Bytes.make(16)
+Bytes.move(0, 8, 8, srcBytes, dstBytes)
+for (let mut i = 8; i < 16; i += 1) {
+  if (i < 8) {
+    assert Bytes.getInt8S(i, dstBytes) == 0l
```

Comment:
What happens to the the bytes in `srcBytes` when you `move` them? Does it shorten that bytes?

Reply:
They stay there. We thought for a while about a name for this one and landed on `move`, which is a nod to C's [`memmove`](https://man7.org/linux/man-pages/man3/memmove.3.html). I do agree that `move` sorta implies that the bytes go away. We think it's far better than OCaml's `blit` 😂 

But obviously open to other suggestions.

Reply:
Hm, so it's more of a `copySlice` or `copySubset`? 🤔

Reply:
Rust seems to call this `copy_to_slice` but they have a slice "type".  Nodejs has optional arguments for their copy method. Additionally, I think this "move" could be replicated with `Bytes.slice` and `Bytes.copy` so I really like `copySlice`

Reply:
One cool thing about `move` is that it can be used to shuffle bytes around within the same `Byte` instance by passing it as both the src and the dest.

Reply:
I think that I still have a preference for `move`, since we already use the "copy" terminology in our libraries to say that we're making a new copy of something and that could add some confusion. Since this utility is largely to move around bytes within a `Bytes`, I think `move` works here!

Reply:
I do think we should assert that those bytes still remain then, but I'm not going to block on it.